pyfr/backends/base/types.py

```diff
@@ -16,34 +16,44 @@ def __init__(self, backend, dtype, ioshape, initval, extent, aliases,
         self.dtype = dtype
         self.itemsize = np.dtype(dtype).itemsize
 
-        # Alignment requirement for the leading dimension
-        ldmod = backend.alignb // self.itemsize if 'align' in tags else 1
-
         # Our shape and dimensionality
         shape, ndim = list(ioshape), len(ioshape)
 
+        # SoA and block column size
+        soasz, csubsz = backend.soasz, backend.csubsz
+        blocked = backend.blocks and 'opmat' not in self.tags
+
         if ndim == 2:
             nrow, ncol = shape
             aosoashape = shape
+
+            # Alignment requirement for the leading dimension
+            ldmod = soasz if 'align' in tags else 1
+            nbcol = csubsz if blocked else ncol - (ncol % -ldmod)
         else:
-            nvar, narr, k = shape[-2], shape[-1], backend.soasz
-            nparr = narr - narr % -k
+            nvar, narr, k = shape[-2], shape[-1], soasz
+            nparr = narr - narr % -csubsz
 
             nrow = shape[0] if ndim == 3 else shape[0]*shape[1]
             ncol = nvar*nparr
             aosoashape = shape[:-2] + [nparr // k, nvar, k]
+            nbcol = nvar*csubsz if blocked else ncol
 
         # Assign
-        self.nrow, self.ncol = int(nrow), int(ncol)
+        self.nrow, self.ncol, self.nbcol = int(nrow), int(ncol), int(nbcol)
```

Comment:
Not sure if these int calls are needed anymore.

Reply:
Removing these int calls caused some issues with cuda kernel with cublas. I got something like wrong argument type again. And when I checked the argument, _as_parameter_ the type was int64. Bringing these int calls back immediately solved the issue. So i'll keep these int calls for now.